### PR TITLE
[cluster-test] Set Autoscaling Group Size from cluster-test

### PR DESCRIPTION
**Pull request by ankushagarwal**
## Summary

* We were seeing issues where an instance was failing to launch in the AWS autoscaling group - causing a libra node to be unassigned
image

* Create `set_asg_size` which sets the autoscaling group size to the min desired capacity plus an additional buffer

* When cluster-test starts, set the autoscaling group size to the desired size plus a buffer of 3 nodes to tolerate upto 3 instance launch failures

This diff depends upon https://github.com/calibra/libra-ops/pull/524

## Test Plan

Ran cluster-test

### CODE REVIEW

File: testsuite/cluster-test/src/main.rs [hunk 1]
```diff
@@ -379,6 +379,22 @@ impl ClusterUtil {
...
+                cluster_swarm
+                    .get_workspace()
+                    .await
+                    .expect("Failed to get workspace")
+            );
+            aws::set_asg_size(
+                (args.k8s_num_validators
+                    + (args.k8s_fullnodes_per_validator * args.k8s_num_validators))
+                    as i64,
+                3,
```
Comment (sausagee) [inline testsuite/cluster-test/src/main.rs:393]: LBT uses 30-node config, and CP uses 100. 3 seems plenty for LBT but not CP.  Should we make the buffer % based instead?

File: scripts/cti [hunk 2]
```diff
@@ -195,6 +195,7 @@ pod_name=${pod_name/_/-} #underscore not allowed in pod name
 specfile=$(mktemp)
 echo "Pod Spec : ${specfile}"
 join_args $*
+ENV="$ENV AWS_ROLE_SESSION_NAME=AWS_ROLE_SESSION_NAME"
```
[the remark this answers is not in the document]
Comment (ankushagarwal) [reply scripts/cti:198]: So this is intentional. AWS_ROLE_SESSION_NAME needs to be set to some non-empty variable, so I've set it to the literal `AWS_ROLE_SESSION_NAME`

File: testsuite/cluster-test/src/main.rs (hunk 1 above)
Comment (ankushagarwal) [reply testsuite/cluster-test/src/main.rs]: I think CP also uses 30-node config. But either way, percent is a good idea. Updated to use `buffer_percent`

**Comment by ankushagarwal** [comment]
The AWS issue still seems to be happening where we request 63 nodes, only 62 of them get provisioned and 1 is stuck in pending. 

Applied calibra/libra-ops#524 to all three clusters and verified that this PR works on all three of them.

**Comment by ankushagarwal** [comment]
So right now, before this PR, autoscaling is managed in the ops repo, but the challenge with that is that we don't get fine grained control. When we launch 60 container, the autoscaler launches exactly 60 nodes.

With this PR, we have more control over what autoscaling looks like and helps us mitigate the AWS issue that we're facing right now

**Comment by andll** [comment]
@bors-libra r+
Thanks for looking into that @ankushagarwal !

File: testsuite/cluster-test/src/main.rs (hunk 1 above)
Comment (andll) [reply testsuite/cluster-test/src/main.rs]: Yep, all tests are using 30 validator + 30 full nodes setup

**Pull request closed without merging**